vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -57,50 +58,112 @@ const (
 // VolumeDriver - Photon volume driver struct
 type VolumeDriver struct {
 	client    *photon.Client
-	hostID    string
-	mountRoot string
+	host      string
 	project   string
-	refCounts *refcount.RefCountsMap
+	tenant    string
 	target    string
+	mountRoot string
+	refCounts *refcount.RefCountsMap
+}
+
+func (d *VolumeDriver) identifyHost(hostIP string) {
+	// Get the list of tenants and projects and identify
+	// the VM ID for this host and its project and tenant
+	// IDs.
+	tenants, err := d.client.Tenants.GetAll()
+	if err != nil {
+		log.WithFields(log.Fields{"target": d.target}).Warning("Invalid target, unable to get list of tenants, exiting.")
+		return
+	}
+	for _, tenant := range tenants.Items {
+		// Get all projects for this tenant
+		projects, err := d.client.Tenants.GetProjects(tenant.ID, nil)
+		if err != nil {
+			log.WithFields(log.Fields{"target": d.target}).Warning("Invalid target, unable to get list of projects, exiting.")
+			return
+		}
+		for _, project := range projects.Items {
+			vms, err := d.client.Projects.GetVMs(project.ID, nil)
+			if err != nil {
+				log.WithFields(log.Fields{"target": d.target}).Warning("Invalid target, unable to fetch list of VMs, exiting.")
+				return
```

Comment:
Even there is an error from GetVMs here, this project.ID may not be the one that the current local node is belong to. So maybe continue this loop instead of return here?